### PR TITLE
183882630 validator details vue

### PR DESCRIPTION
**Pull request by maciejkocylapc**
#### What's this PR do?
- rewride validator details page to vue

#### How should this be manually tested?
- run tests
- check if page looks similar to production
- check if all the links are working

#### What are the relevant tickets?
- [URL to the PivotalTracker ticket](https://www.pivotaltracker.com/story/show/183882630)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

**Review by kamtarnowski — CHANGES_REQUESTED** [review]
Cosmetic changes required.

File: app/javascript/packs/validators/validator_details.vue [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+            class="btn btn-sm btn-secondary"
+            target="_blank">Delegate on Solstake.io</a>
+          <a href="kiwi_url(validator)"
+            title="Delegate SOL to this validator on Staking.kiwi"
+            class="btn btn-sm btn-secondary"
+            target="_blank">Delegate on Staking.kiwi</a>
+          <a href="blazestake_url(validator)"
+            title="Delegate SOL to this validator on BlazeStake"
+            class="btn btn-sm btn-secondary"
+            target="_blank">Delegate on Staking.kiwi</a>
```
Comment (kamtarnowski) [inline app/javascript/packs/validators/validator_details.vue:26]: It should point out to the BlazeStake.

**Review by kamtarnowski — CHANGES_REQUESTED** [review]

**Comment by kamtarnowski** [comment]
Comparing to master branch, some things have been broken.

1)
Zrzut ekranu 2022-12-16 o 13 49 14 (3)
- Delegate button for BlazeStake
- admin warning button (conditional statement is missing)

2) 
Zrzut ekranu 2022-12-16 o 13 49 08 (3)
- within diagrams there is no data for x and y axis
- no horizontal lines
- there are rectangles instead of wheels.

**Comment by kamtarnowski** [comment]
set_continuous_supply_updat
Zrzut ekranu 2022-12-20 o 11 01 55 (3)
Zrzut ekranu 2022-12-20 o 11 02 02 (3)

This detail view is still not valid.

**Comment by kamtarnowski** [comment]
@maciejkocylapc Still there are problems. Please compare it with master branch:

Zrzut ekranu 2022-12-28 o 09 16 15

Master:
Zrzut ekranu 2022-12-28 o 09 23 29

Issues spotted:
* "Skipped Slots History" title missed
* there are rectangles instead of wheels
* no horizontal lines
* no axis description (Dist Behind Leader, Previous X Observations, Percent)
* the range values of x axis are too much detailed, there are too many of them

File: app/javascript/packs/validators/components/block_history_table.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  },
+  methods: {
+    block_stats(history) {
+      return this.block_history_stats.find(o => o.batch_uuid == history.batch_uuid);
+    },
+    history_created_at(history) {
+      return moment(new Date(history.created_at)).utc().format('YYYY-MM-DD HH:mm:ss z')
+    },
+    formatted_number(n) {
+      return n.toLocaleString().replace("&nbsp;", ",")
```
Comment (kbiala) [inline app/javascript/packs/validators/components/block_history_table.js:27]: should be:
```suggestion
      return n.toLocaleString('en-US').replace("&nbsp;", ",")
```

File: app/controllers/validators_controller.rb [hunk 3]
```diff
@@ -45,8 +45,11 @@ def show
     time_from = Time.now - 24.hours
     time_to = Time.now
 
+    @validator.name = @validator.name&.gsub(" ", "-space-")
```
Comment (kbiala) [inline app/controllers/validators_controller.rb:48]: Why are we doing this?
Also, why are we using this method instead of updating API controller?

**Review by kbiala — CHANGES_REQUESTED** [review]
Some validators are failing with `SyntaxError: Unterminated string in JSON`
eg.
http://localhost:3000/validators/6aHUQzbJfX9cb4G6WfBYu7LWjjvEEXak8LHyX2rz7mPM?locale=en&network=mainnet

File: app/javascript/packs/validators/components/block_history_table.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            <td class='text-end'>
+              {{ formatted_number(history.blocks_produced) }}<br />
+              {{ formatted_number(block_stats(history).total_blocks_produced) }}
+            </td>
+            <td class='text-end'>
+              {{ formatted_number(history.skipped_slots) }}<br />
+              {{ formatted_number(block_stats(history).total_slots_skipped) }}
+            </td>
+            <td class='text-end'>
+              {{ (history.skipped_slot_percent * 100.0) }}%<br />
```
Comment (kbiala) [inline app/javascript/packs/validators/components/block_history_table.js:72]: Same here:
`().toLocaleString('en-US', {maximumFractionDigits: 2})`

**Review by kbiala — CHANGES_REQUESTED** [review]

File: app/javascript/packs/validators/components/block_history_table.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+              {{ (block_stats(history).total_slots_skipped / block_stats(history).total_slots * 100.0).toLocaleString(2) }}%
```
Comment (kbiala) [inline app/javascript/packs/validators/components/block_history_table.js:73]: This is not a correct use of options
should be
`.toLocaleString('en-US', {maximumFractionDigits: 2})`

**Review by kbiala — CHANGES_REQUESTED** [review]

File: app/javascript/packs/validators/validator_details.vue [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+              </tr>
+            </tbody>
+          </table>
+        </div>
+      </div>
+    </div>
+
+    <div class="row mb-3">
+      <div class="col-md-9 col-lg-10">
+        <a :href="'/validators?network=' + validator.network + '&order=' + order + '&refresh=' + refresh + '&page=' + page"
```
Comment (kbiala) [inline app/javascript/packs/validators/validator_details.vue:176]: There are a couple of params so I'd extract them to a separate variable, or even full path to a separate method.

File: app/javascript/packs/validators/validator_details.vue [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+           Back to All Validators
+        </a>
+      </div>
+
+      <div class='col-md-3 col-lg-2 toggle-container pt-2' data-turbolinks="false">
+          <a :href="'/validators/' + validator.account + '?network=' + validator.network + '&order=' + order + '&refresh=' + !refresh + '&page=' + page">
+            <i class="fas fa-toggle-on toggle" v-if="refresh"></i>
+            <i class="fas fa-toggle-off toggle" v-if="!refresh"></i>
+          </a>
+          <p class="small text-muted toggle-label" v-if="refresh">REFRESH ON</p>
```
Comment (kbiala) [inline app/javascript/packs/validators/validator_details.vue:187]: Please remove refresh buttons+logic.

**Review by kbiala — CHANGES_REQUESTED** [review]

File: app/javascript/packs/validators/validator_details.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+document.addEventListener('turbolinks:load', () => {
+  new Vue({
+    el: '#validator-details',
+    store,
+    render(createElement) {
+      return createElement(ValidatorDetails, {
+        props: {
+          account: this.$el.attributes.account.value,
+          // score: JSON.parse(this.$el.attributes.score.value),
```
Comment (kbiala) [inline app/javascript/packs/validators/validator_details.js:23]: Comments to remove

**Review by kbiala — APPROVED** [review]

**Pull request merged**